pytket/extensions/azure/backends/azure.py

```diff
@@ -155,13 +191,139 @@ def backend_info(self) -> BackendInfo:
 
     @property
     def required_predicates(self) -> list[Predicate]:
-        return [GateSetPredicate(_ALL_GATES)]
+        return [GateSetPredicate(_ALL_GATES), NoSymbolsPredicate()]
+
+    def _default_2q_gate(device_name: str) -> OpType:
+        return OpType.ZZPhase
 
     def rebase_pass(self) -> BasePass:
-        return AutoRebase(gateset=_GATE_SET)
+        if (
+            self._backendinfo.device_name
+            and self._backendinfo.device_name[:11] == "quantinuum."
+        ):
+            return AutoRebase(
+                _QUANTINUUM_TARGET_GATESET,
+                allow_swaps=True,
+            )
+        else:
+            return AutoRebase(gateset=_GATE_SET)
+
+    def default_compilation_pass(
+        self, optimisation_level: int = 2, timeout: int = 300
+    ) -> BasePass:
+        """
+        :param optimisation_level: Allows values of 0, 1, 2 or 3, with higher values
+            prompting more computationally heavy optimising compilation that
+            can lead to reduced gate count in circuits.
+        :param timeout: Only valid for optimisation level 3, gives a maximimum time
+            for running a single thread of the pass `GreedyPauliSimp`. Increase for
+            optimising larger circuits.
+
+        :return: Compilation pass for compiling circuits to Quantinuum devices
+        """
+        assert optimisation_level in range(4)
+
+        if (
+            self._backendinfo.device_name
+            and self._backendinfo.device_name[:5] == "ionq."
```

Comment:
Are we assuming that ionq and quantinuum are the only backend providers? I think we should have an extensible enum of providers that we switch on here.

Reply:
Like this? [cb3defd](https://github.com/CQCL/pytket-azure/pull/32/commits/cb3defde164fd4471e22d1f6b03261724337a488)